### PR TITLE
Update `vars_funs` interface in the R package to match the Python package

### PR DESCRIPTION
**Pull request by jeancochrane**
This PR branches off of https://github.com/ccao-data/ccao/pull/32 to bring the R package up to parity with the changes that PR makes to the Python package. In particular:

* Remove the unused `vars_check_class` function
* Deprecate the `type` and `dict` params in `vars_rename` and `vars_recode`, and replace them with the new params `{code|output}_type` and `dictionary`

Since [no code is currently using the `vars_check_class` function](https://github.com/search?q=org%3Accao-data%20vars_check_class&type=code), we can safely remove it. Code is still using `type` and `dict`, however, so we implement deprecation with fallback so that that code will still work with the latest version of the package.

Note that there is more code that we could strip out of the R package, but I'm restricting the scope of this PR to only the changes that are necessary to create parity with the Python package as of https://github.com/ccao-data/ccao/pull/32.

### CODE REVIEW

File: R/vars_funs.R
Comment (dfsnow) [inline R/vars_funs.R:1]: **suggestion:** IMO we can also remove the `class_dict` dataset that feeds this function, since it's now just a dbt seed that lives in `data-architecture`. I don't think it's used in any other major repos.

Comment (dfsnow) [inline R/vars_funs.R:1]: **suggestion:** IMO we can also remove the `class_dict` dataset that feeds this function, since it's now just a dbt seed that lives in `data-architecture`. I don't think it's used in any other major repos.

File: R/vars_funs.R
Comment (dfsnow) [inline R/vars_funs.R:93]: **suggestion (non-blocking):** I recognize this is good practice, but I wonder about the utility of including lifecycle code like this in a strictly internal package. My preference would be to just release a new major version with the breaking changes included. Ultimately I defer to you @jeancochrane, as my view here isn't super strongly held.

**Review by dfsnow — APPROVED** [review]
Looks good @jeancochrane 👍 IMO we should just cut a new major version of this package once all the Python functions are merged and R functions updated.

File: R/vars_funs.R
Comment (jeancochrane) [reply R/vars_funs.R]: Makes sense, I stripped out this logic in 3d94d7e7e140851fc1afb9a313a7e9828ef4c905.

File: R/vars_funs.R
Comment (jeancochrane) [reply R/vars_funs.R:1]: Cool, done! I opened https://github.com/ccao-data/data-architecture/pull/667 and https://github.com/ccao-data/ptaxsim/pull/45 to fix the only references I could find outside this repo.

**Pull request merged**
